src/lib/util/FormatMessage.ts

```diff
@@ -1,20 +1,20 @@
 import { Message } from 'discord.js';
-import {
-  languageMappings,
-  languageNameMappings,
-} from '../formatters/FormatterMappings';
+import { languageMappings } from '../formatters/FormatterMappings';
 import { UploadToPastecord } from '../infra/external/pastecordintegration';
 import { reformat } from './reformatter';
 import { asyncStringReplacer, commentify } from './utils';
 
-export const formatMessage = async (message: Message) => {
+export const formatMessage = async (
+  message: Message,
+  _onlyCodeBlocks?: boolean
+) => {
   const content = message.content;
   // The first language key found, it'll be used for the pastecord file and all the comments on the file
-  let firstLanguageKey: string | undefined;
+  let firstLanguageKey: string | undefined = content.match(
+    new RegExp(`\`\`\`(${Object.keys(languageMappings).join('|')})\n`, 'm')
```

Comment:
Instead of assigning it later on, this should be assigned at the very beginning since we want code blocks that come before a valid code block be commented judging by this